src/utils/yaml_node_utils.cpp

```diff
@@ -121,18 +155,25 @@ YAML::Node mp::utils::make_cloud_init_network_config(
     const std::string& file_content)
 {
     YAML::Node network_data = file_content.empty() ? YAML::Node{} : YAML::Load(file_content);
-
     network_data["version"] = "2";
-    network_data["ethernets"]["default"] = create_default_interface_node(default_mac_addr);
 
-    for (size_t i = 0; i < extra_interfaces.size(); ++i)
+    const auto default_interface = interface_details{default_mac_addr};
+    network_data["ethernets"][default_interface.name] = default_interface;
+
+    auto extra_idx = kExtraInterfaceIndexStart;
+    // FIXME: C++20 range-for init statement:
+    // https://en.cppreference.com/w/cpp/language/range-for.html
+    // C++23: https://en.cppreference.com/w/cpp/ranges/enumerate_view.html
+    for (const auto& extra : extra_interfaces)
```

Comment:
Until we can enumerate, I think the traditional for loop remains more adequate for this. Anyway, I am not going to split hairs on this one now.